lib/discover/local.go

```diff
@@ -197,6 +197,21 @@ func (c *localClient) registerDevice(src net.Addr, device Device) bool {
 		}
 
 		if len(tcpAddr.IP) == 0 || tcpAddr.IP.IsUnspecified() {
+			srcAddr, err := net.ResolveTCPAddr("tcp", src.String())
```

Comment:
Or just skip it. Resolve shouldn't talk to DNS as long as the thing given to it looks like an IP, so it doesn't matter much.